File: BaseLibrary/command/handshake/GameTypeCommand.cs
using System.Collections.Generic;

namespace BaseLibrary.command.handshake {
    public class GameTypeCommand : AHandShakeCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        public int ROBOTS_IN_ONE_TEAM { get; private set; }
        public GameType GAME_TYPE { get; private set; }

        public GameTypeCommand(int ROBOTS_IN_ONE_TEAM, GameType GAME_TYPE) {
            this.ROBOTS_IN_ONE_TEAM = ROBOTS_IN_ONE_TEAM;
            this.GAME_TYPE = GAME_TYPE;
        }
    }

    public enum GameType {
        DEADMATCH,
        CAPTURE_FLAG,
        CAPTURE_BASE
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/handshake/GameTypeCommand.java
package BaseLibrary.communication.command.handshake;

import BaseLibrary.communication.protocol.AProtocol;

public class GameTypeCommand extends AHandShakeCommand {

	private static final String NAME = "GAME_TYPE";

	static {
		AProtocol.registerForDeserialize(NAME, GameTypeCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}

	public static enum GameType {
		DEADMATCH, CAPTURE_FLAG, CAPTURE_BASE;
	}

	public final int ROBOTS_IN_ONE_TEAM;
	public final GameType GAME_TYPE;

	public GameTypeCommand(int ROBOTS_IN_ONE_TEAM, GameType GAME_TYPE) {
		this.ROBOTS_IN_ONE_TEAM = ROBOTS_IN_ONE_TEAM;
		this.GAME_TYPE = GAME_TYPE;
	}
}
